server/cmd/code-annotation/handler/assignments.go

```diff
@@ -0,0 +1,74 @@
+package handler
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"strconv"
+
+	"github.com/go-chi/chi"
+	"github.com/go-chi/render"
+
+	"github.com/src-d/code-annotation/server/cmd/code-annotation/serializer"
+	"github.com/src-d/code-annotation/server/model"
+	"github.com/src-d/code-annotation/server/repository"
+)
+
+func HandleGetAssignmentsForUserExperiment(w http.ResponseWriter, r *http.Request) {
+	experimentID, err := strconv.Atoi(chi.URLParam(r, "experimentId"))
+	if err != nil {
+		render.Render(w, r, serializer.NewErrors(err))
+		return
+	}
+
+	userID, err := strconv.Atoi("1")
+	if err != nil {
+		render.Render(w, r, serializer.NewErrors(err))
+		return
+	}
+
+	assignments, err := repository.GetAssignmentsFor(userID, experimentID)
+	if err != nil {
+		render.Render(w, r, serializer.NewErrors(err))
+		return
+	}
+
+	render.Render(w, r, &serializer.ResponseMany{Data: serializer.NewAssignments(assignments)})
+}
+
+func HandleSaveAssignment(w http.ResponseWriter, r *http.Request) {
+	assignmentId, err := strconv.Atoi(chi.URLParam(r, "assignmentId"))
+	if err != nil {
+		render.Render(w, r, serializer.NewErrors(err))
+		return
+	}
+
+	body, err := ioutil.ReadAll(r.Body)
+	defer r.Body.Close()
+	if err != nil {
+		render.Render(w, r, serializer.NewErrors(err))
+		return
+	}
+
+	var assignment serializer.Assignment
+	if err := json.Unmarshal(body, &assignment); err != nil {
+		render.Render(w, r, serializer.NewErrors(err))
+		return
+	}
+
+	answer := model.Answer(assignment.Answer)
+	if err := repository.SaveAssignment(assignmentId, answer, assignment.Duration); err != nil {
```

Comment:
Usually repository save methods accept model. But up to you.

Reply:
Would you like something like


```go
package repository

func UpdateAssignment(id int, answer model.Assignment) error
```
and then use it as below?
```go
if err := repository.UpdateAssignment(assignmentId, assignmentPayload); err != nil {
```

&ast; I send the `assignmentId`, because I'd not rely in the `assignmentPayload`

Reply:
I would prefer.
```go
func UpdateAssignment(model.Assignment) error
```

and in handler or where you create model:
```go
id := getId(r)
payload := decodeAssigment(r.Body)
if err := payload.Validate(); err != nil {
  // handle error
}
a, err := repository.GetAssignment(id)
if err != nil {
  // handle error
}
// could be abstracted to a function
a.Answer = payload.Answer
a.Duration = payload.Duration
// UpdateAssignment doesn't care how "a" was created
repository.UpdateAssignment(a)
```
why would I do like this? Because logic of modifying model belongs to handler. I don't think repository should know anything about payload.

```go
func UpdateAssignment(id int, answer model.Assignment) error
```
has a problem: what if `answer` has `ID` already?
The behavior will be difficult to understand.

But I'm not very strong opinionated about it. If you prefer another style I'm fine with it.